src/openeo_aggregator/backend.py

```diff
@@ -675,74 +675,120 @@ def _get_connection_and_backend_service_id(
         con = self._backends.get_connection(backend_id)
         return con, backend_service_id
 
+    def get_supported_backend_ids(self) -> List[str]:
+        """Get a list containing IDs of the backends that support secondary services."""
+
+        # Try the cache first, but its is cached inside the result of get_service_types().
+        # Getting the service types will also execute querying the capabilities for
+        # which backends have secondary services.
+        return self._get_service_types_cached()["supported_backend_ids"]
+
+    def _find_ids_supported_backends(self) -> List[str]:
+        """Query the backends capabilities to find the ones that support secondary services."""
+        return [
+            con.id
+            for con in self._backends
+            if con.capabilities().supports_endpoint("/service_types")
+        ]
+
     def service_types(self) -> dict:
         """https://openeo.org/documentation/1.0/developers/api/reference.html#operation/list-service-types"""
-        cached_info = self._get_service_types_cached()
+        service_types = self._get_service_types_cached()["service_types"]
         # Convert the cached results back to the format that service_types should return.
-        return {name: data["service_type"] for name, data, in cached_info.items()}
+        return {name: data["service_type"] for name, data, in service_types.items()}
+
+    def _get_service_types_cached(self):
+        return self._memoizer.get_or_call(
+            key=("service_types"), callback=self._get_service_types
```

Comment:
Bit misleading: the parenthesis around `("service_types")` make it look like a tuple, while it just resolves to a string

Reply:
Indeed, a mistake and misleading.
Corrected it.